src/testing/builder/CategoriesMapModelBuilder.js

```diff
@@ -0,0 +1,108 @@
+    this._maxWidth = maxWidth
+    this._depth = depth
+  }
+
+  predictableNumber (index: number, max: number = MAX_PREDICTABLE_VALUE): number {
+    return seedrandom(`${index}-seed`)() * max
+  }
+
+  createResource (url: string, index: number, lastUpdate: moment): FileCacheStateType {
+    const hash = hashUrl(url)
+    return {
+      [url]: {
+        filePath: `path/to/documentDir/resource-cache/v1/some-city/files/${hash}.png`,
+        lastUpdate: moment(lastUpdate).add(this.predictableNumber(index), 'days'),
+        hash
+      }
+    }
+  }
+
+  addChildren (
+    category: CategoryModel,
+    categories: Array<CategoryModel>,
+    resourceCache: LanguageResourceCacheStateType,
+    depth: number) {
+    if (depth === this._depth) {
+      return
+    }
+
+    for (let i = 0; i < this._maxWidth; i++) {
+      const path = `${category.path}/category_${depth}-${i}`
+      const lastUpdate = moment('2017-11-18T19:30:00.000Z', moment.ISO_8601)
+
+      const newChild = new CategoryModel({
+        id: 0,
+        path,
+        title: `Category ${depth}-${i}`,
+        content: `<h1>This is a sample page</h1>
+                    <img src="https://integreat/title_${depth}-${i}-300x300.png"/>
+                    <p>This is a sample page</p>
+                    <img src="https://integreat/category_${depth}-${i}-300x300.png"/>`,
+        order: -1,
+        availableLanguages: new Map(),
+        thumbnail: `http://thumbnails/category_${depth}-${i}.png`,
+        parentPath: category.path,
+        lastUpdate
+      })
+
+      resourceCache[path] = {
+        ...this.createResource(`https://integreat/title_${depth}-${i}-300x300.png`,
+          depth * this._maxWidth + i,
+          lastUpdate),
+        ...this.createResource(`https://integreat/category_${depth}-${i}-300x300.png`,
+          depth * this._maxWidth + i,
+          lastUpdate)
+      }
+      categories.push(newChild)
+      this.addChildren(newChild, categories, resourceCache, depth + 1)
+    }
+  }
+
+  buildResources (): { [path: string]: FileCacheStateType } {
+    return this.buildAll().resourceCache
+  }
+
+  build (): CategoriesMapModel {
+    return this.buildAll().categories
+  }
+
+  buildAll (): { categories: CategoriesMapModel, resourceCache: { [path: string]: FileCacheStateType } } {
+    const categories = []
+    const resourceCache = {}
+
+    for (let i = 0; i < this._maxWidth; i++) {
+      this.addChildren(new CategoryModel({
+        id: 0,
+        path: '/augsburg/de',
+        title: 'augsburg',
+        content: '',
+        order: -1,
+        availableLanguages: new Map(),
+        thumbnail: 'no_thumbnail',
+        parentPath: '',
+        lastUpdate: moment('2017-11-18T19:30:00.000Z', moment.ISO_8601)
+      }), categories, resourceCache, 0)
+    }
+
+    return { categories: new CategoriesMapModel(categories), resourceCache }
+  }
+}
+
+export default CategoriesMapModelBuilder
```

Comment:
I'd actually suggest testing this, since it's quite complex. A single snapshot might do the trick.

Reply:
You are right, I created a task: https://issues.integreat-app.de/browse/NATIVE-350